services/scan/src/store.ts

```diff
@@ -946,6 +952,8 @@ export class Store {
         writeStream.write('\n');
       }
     }
+
+    writeStream.end();
```

Comment:
One thing to note is that we're not really using the writable stream API correctly. This can lead to excessive buffering and potentially a much slower export operation due to the memory usage/GC/swapping going on. See https://enlear.academy/nodejs-backpressuring-in-streams-52638f505e1b for what we should be doing differently, specifically waiting for a `drain` event after each `write` that returns `false`.

Reply:
Interesting. I'm not sure exactly how it will look - I tried just waiting on the writeStream to drain but we can't do that because we have an ongoing SQL query that we're iterating through, so may have to change the way we're working through the data. Not going to touch that for now.